fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerMutualTest.java

```diff
@@ -976,4 +978,196 @@ void testMultiTargetMultiType() {
         });
         validateIndexes(indexes);
     }
+
+    @Test
+    void testMutualIndexingBlocker() {
```

Comment:
Each type of index build gets a different set of tests. They're all testing the block logic, which should mostly be orthogonal to the type of index build.
The code doesn't have the right abstractions so that we can test this independently from the type of index build, and changing it, i think, would be a fairly sizeable undertaking, that we shouldn't do now.

That being said, perhaps it would make sense to create an abstract test class for this, with some overrides for setting up and doing the different build attempts that could test all the behavior you have tests for and run it for all the different types of index builds.

Reply:
That makes sense. One of the things in my todo list is to restructure the tests derived from `OnlineIndexerTest` and share common code in the base module.